src/goto-instrument/contracts/dynamic-frames/dfcc_infer_loop_assigns.cpp

```diff
@@ -154,10 +157,75 @@ std::unordered_set<irep_idt> gen_loop_locals_set(
   return loop_locals;
 }
 
-assignst dfcc_infer_loop_assigns(
+/// Append all identifiers in `expr` into `identifiers`.
+static void
+append_ids_in_expr(const exprt &expr, std::unordered_set<irep_idt> &identifiers)
+{
+  auto ids = find_symbol_identifiers(expr);
+  identifiers.insert(ids.begin(), ids.end());
+}
+
+/// Find all identifiers in `src`.
+static std::unordered_set<irep_idt>
+find_symbol_identifiers(const goto_programt &src)
+{
+  std::unordered_set<irep_idt> identifiers;
+  forall_goto_program_instructions(i_it, src)
```

Comment:
I'd suggest a ranged for `for(const auto &instruction : src.instructions)` to both avoid the macro and the unnecessary iterator.